iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -676,6 +678,32 @@ public void executeOperation(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
     }
   }
 
+  @Override
+  public void createBranchOperation(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
+      AlterTableCreateBranchSpec createBranchSpec) {
+    TableDesc tableDesc = Utilities.getTableDesc(hmsTable);
+    Table icebergTable = IcebergTableUtil.getTable(conf, tableDesc.getProperties());
+
+    String branchName = createBranchSpec.getBranchName();
+    ManageSnapshots manageSnapshots = icebergTable.manageSnapshots();
+    Long snapShotId = Optional.ofNullable(createBranchSpec.getSnapshotId())
+        .orElse(icebergTable.currentSnapshot().snapshotId());
```

Comment:
What happens to this in case of any empty table? Just a create iceberg table and nothing. As far as I remember in that case currentSnapshot is null? Just give a check once.

Reply:
Nice catch! fixed. Thanks very much!